src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1010,17 +1001,18 @@ def delete_application(client, identifier):
 
 def _resolve_application(client, identifier):
     """Resolve an application and return its id."""
-    result = client.application_list(filter="identifierUris/any(s:s eq '{}')".format(identifier))
-    if not result:
-        if is_guid(identifier):
-            # it is either app id or object id, let us verify
-            result = client.application_list(filter="appId eq '{}'".format(identifier))
-        else:
-            error = CLIError("Application '{}' doesn't exist".format(identifier))
+    if is_guid(identifier):
+        # it is either app id or object id, let us verify
+        result = client.application_list(filter="appId eq '{}'".format(identifier))
+        # If not found, this looks like an object id
+        return result[0][ID] if result else identifier
+    else:
+        result = client.application_list(filter="identifierUris/any(s:s eq '{}')".format(identifier))
+        if not result:
+            error = CLIError("Application with identifier URI '{}' doesn't exist".format(identifier))
             error.status_code = 404  # Make sure CLI returns 3
             raise error
-
-    return result[0][ID] if result else identifier
+        return result[0][ID]
```

Comment:
According to https://docs.microsoft.com/en-us/azure/active-directory/develop/security-best-practices-for-app-registration#appid-uri-configuration, identifier URI must look like `api://` or `https://`. 

If `identifier` is a GUID, there is no need to treat it as `identifierUris` to get the application. This refactor removes the unnecessary REST API invocation to get the applications with `identifier` as one of `identifierUris`. This reduces the execution time of application resolution by 50%.

Con: It is better to re-record all existing recordings involving application resolution.